plugins/outputs/sensu/sensu.go

```diff
@@ -0,0 +1,468 @@
+		// Add tags from config to each metric point
+		var tagList []*OutputTag
+		for name, value := range s.Tags {
+			metric.AddTag(name, value)
+		}
+		for _, tagSet := range metric.TagList() {
+			tag := &OutputTag{
+				Name:  tagSet.Key,
+				Value: tagSet.Value,
+			}
+			tagList = append(tagList, tag)
+		}
+
+		// Get all valid numeric values, convert to float64
+		for _, fieldSet := range metric.FieldList() {
+			key := fieldSet.Key
+			// don't need err since math.NaN is returned with err
+			value, _ := getFloat(fieldSet.Value)
+			// JSON does not support these special values
+			if math.IsNaN(value) || math.IsInf(value, 0) {
+				continue
+			}
+
+			point := &OutputMetric{
+				Name:      metric.Name() + "." + key,
+				Tags:      tagList,
+				Timestamp: metric.Time().Unix(),
+				Value:     value,
+			}
+			points = append(points, point)
+		}
+	}
+
+	reqBody, err := s.Wrapper(points)
+
+	if err != nil {
+		return err
+	}
+
+	if err := s.write(reqBody); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *Sensu) write(reqBody []byte) error {
+	var reqBodyBuffer io.Reader = bytes.NewBuffer(reqBody)
+	method := http.MethodPost
+
+	var err error
+	if s.ContentEncoding == "gzip" {
+		rc, err := internal.CompressWithGzip(reqBodyBuffer)
+		if err != nil {
+			return err
+		}
+		defer rc.Close()
+		reqBodyBuffer = rc
+	}
+
+	endpointUrl, err := s.GetEndpointUrl()
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest(method, endpointUrl, reqBodyBuffer)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("User-Agent", internal.ProductToken())
+
+	req.Header.Set("Content-Type", defaultContentType)
+	if s.ContentEncoding == "gzip" {
+		req.Header.Set("Content-Encoding", "gzip")
+	}
+
+	if s.ApiKey != nil {
+		bearerType := "Bearer"
+		if s.BackendApiUrl != nil {
+			bearerType = "Key"
+		}
+		req.Header.Set("Authorization", bearerType+" "+*s.ApiKey)
+	}
+
+	for k, v := range s.Headers {
+		if strings.ToLower(k) == "host" {
+			req.Host = v
+		}
+		req.Header.Set(k, v)
+	}
+
+	resp, err := s.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	_, err = ioutil.ReadAll(resp.Body)
+
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		return fmt.Errorf("when writing to [%s] received status code: %d", endpointUrl, resp.StatusCode)
+	}
+
+	return nil
+}
+
+// Resolves the event write endpoint
+func (s *Sensu) GetEndpointUrl() (string, error) {
+	var endpointUrl string
+
+	if s.BackendApiUrl != nil {
+		endpointUrl = *s.BackendApiUrl
+	} else if s.AgentApiUrl != nil {
+		endpointUrl = *s.AgentApiUrl
+	}
+
+	u, err := url.Parse(endpointUrl)
+	if err != nil {
+		return endpointUrl, err
+	}
+
+	if s.BackendApiUrl != nil {
+		namespace := "default"
+		if s.Entity != nil && s.Entity.Namespace != nil {
+			namespace = *s.Entity.Namespace
+		}
+		u.Path = path.Join(u.Path, "/api/core/v2/namespaces", namespace, "events")
+	} else {
+		u.Path = path.Join(u.Path, "/events")
+	}
```

Comment:
Determine the "path_suffix" in the `if` above and then just do
```suggestion
	u.Path = path.Join(u.Path, path_suffix)
```
here. This way you keep the special handling of the URL options in one place.

Reply:
👍